From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benonardo <benonardo@icloud.com>
Date: Fri, 22 Oct 2021 19:48:17 +0200
Subject: [PATCH] New server name


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 49821f441886b4de2cac5b1b13981e2161992f98..f7612ec6ddf96045ec5ac0280422fd8e58371326 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1726,7 +1726,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "Airplane"; // Airplane // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "BenPlane"; // BenPlane // Airplane // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
