From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benonardo <benonardo@icloud.com>
Date: Sun, 24 Oct 2021 19:45:36 +0200
Subject: [PATCH] Removed keepalive timeout


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 54b837d3a70862622a3e6d2a5f900eff4ead9103..af337cdd6c422e106ed02e491938af0ed8d8431d 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -389,10 +389,12 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
         // Purpur end
 
         if (this.keepAlivePending) {
-            if (!this.processedDisconnect && elapsedTime >= KEEPALIVE_LIMIT) { // check keepalive limit, don't fire if already disconnected
-                ServerGamePacketListenerImpl.LOGGER.warn("{} was kicked due to keepalive timeout!", this.player.getScoreboardName()); // more info
-                this.disconnect(new TranslatableComponent("disconnect.timeout", new Object[0]), org.bukkit.event.player.PlayerKickEvent.Cause.TIMEOUT); // Paper - kick event cause
-            }
+            // BenPlane start
+            //if (!this.processedDisconnect && elapsedTime >= KEEPALIVE_LIMIT) { // check keepalive limit, don't fire if already disconnected
+            //    ServerGamePacketListenerImpl.LOGGER.warn("{} was kicked due to keepalive timeout!", this.player.getScoreboardName()); // more info
+            //    this.disconnect(new TranslatableComponent("disconnect.timeout", new Object[0]), org.bukkit.event.player.PlayerKickEvent.Cause.TIMEOUT); // Paper - kick event cause
+            //}
+            // BenPlane end
         } else {
             if (elapsedTime >= 15000L) { // 15 seconds
                 this.keepAlivePending = true;
