From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benonardo <benonardo@icloud.com>
Date: Sat, 23 Oct 2021 20:46:11 +0200
Subject: [PATCH] Added BenPlane package


diff --git a/src/main/java/com/benonardo/benplane/BenPlaneServer.java b/src/main/java/com/benonardo/benplane/BenPlaneServer.java
new file mode 100644
index 0000000000000000000000000000000000000000..c948290e536225e6db71e0c0ad05ada24c7421a0
--- /dev/null
+++ b/src/main/java/com/benonardo/benplane/BenPlaneServer.java
@@ -0,0 +1,20 @@
+package com.benonardo.benplane;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class BenPlaneServer {
+
+    public static BenPlaneServer instance;
+    private final Logger logger = LogManager.getLogger("BenPlane");
+
+    public BenPlaneServer() {
+        instance = this;
+        instance.onStart();
+    }
+
+    private void onStart() {
+        logger.info("BenPlane started");
+    }
+
+}
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index f7612ec6ddf96045ec5ac0280422fd8e58371326..995edae59800b4eea177153361c7364bc9b05e6e 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import com.benonardo.benplane.BenPlaneServer;
 import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 import co.aikar.timings.Timings;
@@ -209,6 +210,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
     private static final int START_TICKING_CHUNK_COUNT = 441;
     private static final int AUTOSAVE_INTERVAL = 6000;
     private static final int MAX_TICK_LATENCY = 3;
+    public BenPlaneServer benPlane;
     public static final int ABSOLUTE_MAX_WORLD_SIZE = 29999984;
     public static final LevelSettings DEMO_SETTINGS = new LevelSettings("Demo World", GameType.SURVIVAL, false, Difficulty.NORMAL, false, new GameRules(), DataPackConfig.DEFAULT);
     private static final long DELAYED_TASKS_TICK_EXTENSION = 50L;
@@ -473,6 +475,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         */
         // Paper end
         Runtime.getRuntime().addShutdownHook(new org.bukkit.craftbukkit.util.ServerShutdownThread(this));
+        this.benPlane = new BenPlaneServer(); // BenPlane
     }
     // CraftBukkit end
 
